ssl/quic/quic_local.h

```diff
@@ -119,18 +120,25 @@ struct quic_xso_st {
     int                             last_error;
 };
 
+/*
+ * QUIC connection SSL object (QCSO) type. This implements the API personality
+ * layer for QCSO objects, wrapping the QUIC-native QUIC_CHANNEL object.
+ */
 struct quic_conn_st {
     /*
-     * ssl_st is a common header for ordinary SSL objects, QUIC connection
-     * objects and QUIC stream objects, allowing objects of these different
-     * types to be disambiguated at runtime and providing some common fields.
+     * QUIC_OBJ is a common header for QUIC APL objects, allowing objects of
+     * these different types to be disambiguated at runtime and providing some
+     * common fields.
      *
      * Note: This must come first in the QUIC_CONNECTION structure.
      */
-    struct ssl_st                   ssl;
+    QUIC_OBJ                        obj;
 
     SSL                             *tls;
 
+    /* The QLSO this connection belongs to, if any. */
+    QUIC_LISTENER                   *listener;
+
```

Comment:
A `quic_conn_st` is never a listener. But it may have been created by a listener, in which case `listener` is non-NULL. If a `quic_conn_st` (`QUIC_CONNECTION`) was created directly then `listener` is NULL, yes.